python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -331,22 +309,25 @@ def __getitem__(self, this_id):
             this_id (int): This is the key of an object in the object dictionary
         """
         logger.debug("__getitem__")
+        # Is this an Incomplete Object?
+        if this_id in self._incomplete_objects:
+            return IncompleteObject(self, this_id)
+
+        # Nope, so try to find it and raise an exception if not
         try:
             return self._objects[this_id]
         except KeyError as err:
             logger.debug("Repo KeyError: %s" % err)
             logger.debug("Keys: %s id: %s" % (self._objects.keys(), this_id))
-            if this_id in self._incomplete_objects:
```

Comment:
The flow is something like this:

 1) Ganga loads index (if it can't load the index or tries to load the job here because the index is corrupt the job is marked as incomplete and placed in `_incomplete_objects`.
  If the job is __not__ loaded here it is still in `_objects`
 2) The user (in some form) tried to load the job from disk. If this completed the object in `_objects` is removed. If it is not the object is now added to `_incomplete_objects`.
  (It may or may not still reside in the `_objects` list but I think it should be replaced by an an `IncompleteObject` class if it does).

At step 2 you cannot always know in advance if the job can be successfully loaded. You can have a job which depends on a plugin which isn't loaded at the moment and this will get into this state.

I see this a LOT due to being careless when moving from LHCb<->LSST and I'm keen not to start obliterating jobs or poisoning sessions with errors.